packages/experimental/Expander/windows/ReactNativeExpander/ExpanderView.cpp

```diff
@@ -0,0 +1,255 @@
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnCollapsing(sender, args);
+            }
+        });
+        m_expanderExpandingRevoker = expander.Expanding(winrt::auto_revoke,
+            [ref = get_weak()](auto const& sender, auto const& args) {
+            if (auto self = ref.get()) {
+                self->OnExpanding(sender, args);
+            }
+        });
+    }
+
+    void ExpanderView::UpdateProperties(winrt::IJSValueReader const& reader) {
+        m_updating = true;
+        auto const& propertyMap = JSValueObject::ReadFrom(reader);
+        auto expander = (this->Content()).try_as<Microsoft::UI::Xaml::Controls::Expander>();
+
+        for (auto const& pair : propertyMap) {
+            auto const& propertyName = pair.first;
+            auto const& propertyValue = pair.second;
+
+            if (propertyName == "expandDirection") {
+                if (propertyValue.IsNull()) {
+                    expander.ClearValue(Microsoft::UI::Xaml::Controls::Expander::ExpandDirectionProperty());
+                }
+                else {
+                    auto direction = propertyValue.AsString();
+                    if (direction == "up") {
+                        expander.ExpandDirection(Microsoft::UI::Xaml::Controls::ExpandDirection::Up);
+                    }
+                    else {
+                        expander.ExpandDirection(Microsoft::UI::Xaml::Controls::ExpandDirection::Down);
+                    }
+                }
+            }
+            else if (propertyName == "expanded") {
+                if (propertyValue.IsNull()) {
+                    expander.ClearValue(Microsoft::UI::Xaml::Controls::Expander::IsExpandedProperty());
+                    m_expanded = false;
+                }
+                else if (expander.IsExpanded() != propertyValue.AsBoolean()) {
+                    m_expanded = propertyValue.AsBoolean();
+                    expander.IsExpanded(m_expanded);
+                }
+            }
+            else if (propertyName == "enabled") {
+                if (propertyValue.IsNull()) {
+                    expander.IsEnabled(true);
+                }
+                else {
+                    expander.IsEnabled(propertyValue.AsBoolean());
+                }
+            }
+            else if (propertyName == "width") {
+                if (!propertyValue.IsNull()) {
+                    expander.Width(propertyValue.AsDouble());
+                    this->Width(propertyValue.AsDouble());
+                }
+            }
+            else if (propertyName == "height") {
+                if (!propertyValue.IsNull()) {
+                    expander.Height(propertyValue.AsDouble());
+                    this->Height(propertyValue.AsDouble());
+                }
+            }
+            else if (propertyName == "contentHorizontalAlignment") {
+                if (!propertyValue.IsNull()) {
```

Comment:
for some of the properties you clear the value if the incoming argument is null - should that be done for all props, or not really?

Reply:
I'll update it to clear the values for properties with associated dependency properties (height, width, alignment). I'm not sure if there's a clearValue equivalent for properties set through the resource dictionary